segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorage.java

```diff
@@ -600,9 +600,29 @@ public boolean supportsAtomicWrites() {
         return true;
     }
 
+    /**
+     * Lists all the segments stored on the storage device.
+     * This method looks only at the segment metadata committed to the underlying {@link ChunkMetadataStore}.
+     *
+     * @return A CompletableFuture that, when completed, will contain an {@link Iterator} that can be used to enumerate and retrieve properties of all the segments.
+     * If the operation failed, it will contain the cause of the failure.
+     */
     @Override
-    public Iterator<SegmentProperties> listSegments() {
-        throw new UnsupportedOperationException("listSegments is not yet supported");
+    public CompletableFuture<Iterator<SegmentProperties>> listSegments() {
+        return metadataStore.getAllEntries()
```

Comment:
How many entries could be "all"? How long can be the list of `SegmentProperties` be?